### PR TITLE
webapp: Adding E2E tests for az webapp up

### PR DESCRIPTION
**Pull request by panchagnula**
Adding test  for az webapp up for Node app

Adding test for Python App for az webapp up

Adding test for dotnercore for az webapp up

Adding tests for static html

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by panchagnula** [comment]
Not updating history.rst since this is just adding tests.

**Comment by panchagnula** [comment]
@ammanifold can you review as well? Thanks!

**Comment by yungezz** [comment]
hi @qwordy  could you pls help to review the PR also? thanks.

**Comment by panchagnula** [comment]
@qwordy can you help me figure out why these tests are failing in the Travis Build?
The command creates a RG & ASP and is this something that is not supported in the CLI test infrastructure? Thanks!

**Comment by qwordy** [comment]
> 
> 
> @qwordy can you help me figure out why these tests are failing in the Travis Build?
> The command creates a RG & ASP and is this something that is not supported in the CLI test infrastructure? Thanks!

Azure CLI testing framework doesn't has such limitation according to my experience.

**Comment by qwordy** [comment]
I am curious about how to test whether the web app created by `az webapp up` is really working. Maybe send a request to the web app in the test function, and check the return code and return content?

**Comment by haroldrandom** [comment]
Hi @panchagnula, I haven't dive down into your code.

Could you make a try on your local machine?
1. Run test with --live and refresh recording YAML file
2. Run test right again in non-live mode (without --live) to see whether it will fail or not

The purpose is to make sure command `az webapp up` is idempotence under the same condition.

**Comment by ammanifold** [comment]
Sorry can you explain why the zip files are needed?

**Comment by panchagnula** [comment]
> I am curious about how to test whether the web app created by `az webapp up` is really working. Maybe send a request to the web app in the test function, and check the return code and return content?

Hmm that's what we do - after the webapp is created we query the GET call & validate. I can make the test liveOnly & it will work but i want these run all the time not just as live tests.

**Comment by panchagnula** [comment]
> Sorry can you explain why the zip files are needed?

az webapp up creates & deploys code - these zips are test code for Node, Python, DotNet test apps we check these in as Zip files & code is extracted during the test to run the command from the DIR where the code is located locally.

**Comment by ammanifold** [comment]
> > Sorry can you explain why the zip files are needed?
> 
> az webapp up creates & deploys code - these zips are test code for Node, Python, DotNet test apps we check these in as Zip files & code is extracted during the test to run the command from the DIR where the code is located locally.

OK gotcha. For some reason I thought we already had these for az webapp create

**Comment by haroldrandom** [comment]
> Hi @panchagnula, I haven't dive down into your code.
> 
> But could you make a try on your local machine?
> 
> 1. Run test with --live and refresh recording YAML file
> 2. Run test right again in non-live mode (without --live) to see whether it will fail or not
> 
> The purpose is to make sure command `az webapp up` is idempotence under the same condition.

Since you have marked those tests `live_only`, then I have no question.

**Comment by panchagnula** [comment]
@qwordy and @Juliehzl  can you take a look at this PR please? Thanks!

**Comment by yonzhan** [comment]
@qwordy @Juliehzl please merge this PR by this week.

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_up_commands.py [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+import os
+
+from azure.cli.testsdk import (
+    ScenarioTest, ResourceGroupPreparer, JMESPathCheck, live_only)
+
+TEST_DIR = os.path.abspath(os.path.join(os.path.abspath(__file__), '..'))
+
+
+class WebAppUpE2ETests(ScenarioTest):
+    @live_only()
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_up_commands.py]: @panchagnula may I know that why you make all your test live_only()? In this way, CI system still will skip the test.

**Pull request merged**